src/plugins/discover/public/application/view_components/canvas/top_nav.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useEffect, useState } from 'react';
+import { i18n } from '@osd/i18n';
+import { AppMountParameters } from '../../../../../../core/public';
+import { NEW_DISCOVER_APP, PLUGIN_ID } from '../../../../common';
+import { useOpenSearchDashboards } from '../../../../../opensearch_dashboards_react/public';
+import { DiscoverServices } from '../../../build_services';
+import { IndexPattern } from '../../../opensearch_dashboards_services';
+
+export interface TopNavProps {
+  opts: {
+    setHeaderActionMenu: AppMountParameters['setHeaderActionMenu'];
+  };
+}
+
+export const TopNav = ({ opts }: TopNavProps) => {
+  const {
+    services: {
+      uiSettings,
+      navigation: {
+        ui: { TopNavMenu },
+      },
+      core: {
+        application: { navigateToApp },
+      },
+      data,
+    },
+  } = useOpenSearchDashboards<DiscoverServices>();
+  const [indexPatterns, setIndexPatterns] = useState<IndexPattern[] | undefined>(undefined);
+
+  useEffect(() => {
+    if (uiSettings.get(NEW_DISCOVER_APP) === false) {
+      const path = window.location.hash;
```

Comment:
I noticed this in the other unchanged code, but why do call the hash a path? We could just use `const hash` and avoid any confusion.